python/gps/agent/mjc/agent_mjc.py

```diff
@@ -99,6 +94,44 @@ def _setup_world(self, filename):
                                        cam_pos[0], cam_pos[1], cam_pos[2],
                                        cam_pos[3], cam_pos[4], cam_pos[5])
 
+    def reset_initial_x0(self, condition):
+        """
+        Reset initial x0 randomly based on sampling_range_x0
+        and prohibited_ranges_x0
+        Args:
+            condition: Which condition setup to run.
+        """
+        self._hyperparams['x0'][condition] = sample_params(
+                self._hyperparams['sampling_range_x0'],
+                self._hyperparams['prohibited_ranges_x0'])
+
+    def reset_initial_body_offset(self, condition):
+        """
+        Reset initial body offset randomly based on sampling_range_bodypos
+        and prohibited_ranges_bodypos
+        Args:
+            condition: Which condition setup to run.
+        """
+        tmp_body_pos_offset = self._hyperparams['pos_body_offset'][condition][:]
```

Comment:
nit: we are now inconsistent between pos_body and bodypos. Maybe we should change pos_body_offset to bodypos_offset?